### PR TITLE
Pokemon Red/Blue: Add new options to slot_data

### PR DESCRIPTION
**Pull request by coveleski**
## What is this fixing or adding?
Added require_pokedex, blind_trainers, and area_1_to_1 mapping, which would be helpful to the poptracker pack to more accurately reflect the checks available to players.

## How was this tested?
Generated a game with 10 games, and checked the spoiler log against the info in slot_data

## If this makes graphical changes, please attach screenshots.
No visual changes

### CODE REVIEW

**Comment by remyjette** [comment]
oops
tried to assign @Alchav but misclicked
anyway there's your ping :)

**Comment by NewSoupVi** [comment]
Happy to merge this quickly before next version, if Alchav approves it.

**Review by Alchav — APPROVED** [review]

**Review by NewSoupVi — APPROVED** [review]

**Pull request merged**
